spot_driver_cpp/src/api/default_image_client_api.cpp

```diff
@@ -224,9 +198,12 @@ tl::expected<std::vector<geometry_msgs::msg::TransformStamped>, std::string> get
 }  // namespace
 
 namespace spot_ros2 {
-DefaultImageApi::DefaultImageApi(std::shared_ptr<Robot> robot) : robot_{robot} {}
 
-tl::expected<GetImagesResult, std::string> DefaultImageApi::getImages(::bosdyn::api::GetImageRequest request) {
+DefaultImageClientApi::DefaultImageClientApi(::bosdyn::client::ImageClient* image_client,
+                                             std::shared_ptr<TimeSyncApi> time_sync_api, const std::string& robot_name)
+    : image_client_{std::move(image_client)}, time_sync_api_{time_sync_api}, robot_name_{robot_name} {}
```

Comment:
You are passing a raw pointer into a `unique_ptr`. This means that it is now your responsibility to destroy the client, is this intentional?  The use of `std::move` is not necessary for raw pointers. `std::move` means that you take the guts out of an object and throw them into another object, so the source object becomes unusable. But a raw pointer has no guts.